co2calculator/calculate.py

```diff
@@ -764,3 +663,105 @@ def commuting_emissions_group(
     group_co2e = aggr_co2 / n_participants * n_members
 
     return group_co2e
+
+
+class InvalidSpatialInput(Exception):
+    """Raised when consumer inputs invalid spatial information"""
+
+
+def _get_distance(start, destination, transportation_mode):
+    """Get the distance between start and destination
+
+    Raises:
+    - InvalidSpatialInput if start and stop are malformed or None
+    """
+
+    # TODO: Refactor to meet DRY
+
+    if None in [start, destination]:
+        raise InvalidSpatialInput("neither distance or start/destination  provided")
+
+    if transportation_mode == "car":
+        # Stops are formatted like:
+        # [
+        #   {
+        #     "address": "Im Neuenheimer Feld 348",
+        #     "locality": "Heidelberg",
+        #     "country": "Germany"
+        #   },
+        #   {
+        #     "country": "Germany",
+        #     "locality": "Berlin",
+        #     "address": "Alexanderplatz 1"
+        #   }
+        # ]
+        # TODO: Validate with BaseModel
+
+        coords = []
+        for loc in [start, destination]:
+            _, _, loc_coords, _ = geocoding_structured(loc)
+            coords.append(loc_coords)
+        return get_route(coords, "driving-car")
+
+    if transportation_mode == "bus":
+        # TODO: Validate with BaseModel
+        # TODO: Question: Why are we not calculating the bus trip like `driving-car` routes?
```

Comment:
I think we are assuming line buses here, which do not go the fastest way between places and that's why we rather take the haversine + detour approach.. with a detour value we found in the literature

Although I'm not completely convinced. For travel buses which are not line buses, get_route would be perfectly fine... Or also get_route + a detour could be an option; OR: Take the example of flixbus: Users could also add the intermediate stops between their start and destination to approximate the route (with get_route)

So I'd say let's discuss this in a next meeting ;)

Reply:
#82